client/src/containers/AuthContainer.test.tsx

```diff
@@ -91,119 +89,155 @@ describe("verifying valid input for signup page", () => {
+
+        const fields = {
+          emailForm: "daniel.zingaro@mail.utoronto.ca",
+          userNameForm: "zingarod",
+          passwordForm: "password",
+          confirmPassForm: "password",
+          firstNameForm: "Daniel",
+          lastNameForm: "Zingaro",
+        };
+        Object.keys(fields).forEach((key) => {
+          fireEvent.change(
+            screen.getByTestId(key).querySelector("input") as HTMLInputElement,
+            { target: { value: fields[key as keyof typeof fields] } }
+          );
+        });
+
+        const form = screen.getByTestId("form");
+        const mockSignUp = jest
+          .spyOn(ServerApi.prototype, "signUp")
+          .mockImplementation(() =>
+            Promise.resolve({ response: {} as AxiosResponse })
+          );
+        const formData = {
+          email: "daniel.zingaro@mail.utoronto.ca",
+          userName: "zingarod",
+          password: "password",
+          firstName: "Daniel",
+          lastName: "Zingaro",
+        };
+
+        await act(async () => {
+          fireEvent.submit(form);
+          await new Promise((r) => setTimeout(r, 1));
+        });
+        expect(mockSignUp).toBeCalledWith(formData);
+
+      });
     });
   });
 });
 
-describe("Login component", () => {
+describe("SignIn component", () => {
   describe("handleSubmit function", () => {
-    it("should show backend error alert on backend error", async () => {
-      screen.getByTestId("LogInTabButton").click();
-
-      const form = screen.getByTestId("form");
-      const msg = "test";
-      const mockSignIn = jest
-        .spyOn(ServerApi.prototype, "signIn")
-        .mockImplementation(() =>
-          Promise.resolve({
-            error: { response: { data: { message: msg } } } as AxiosError,
-          })
-        );
-      const mockAlert = jest
-        .spyOn(window, "alert")
-        .mockImplementation(() => {});
-
-      fireEvent.submit(form);
-      await new Promise((r) => setTimeout(r, 1));
-      expect(mockSignIn).toBeCalled();
-      expect(mockAlert).toBeCalledWith(`Message: ${msg}`);
-    });
+    describe("On error", () => {
+      it("should not call SignIn if form is missing fields", async () => {
+        screen.getByTestId("SignInTabButton").click();
+
+        const form = screen.getByTestId("form");
+        const mockSignIn = jest
+          .spyOn(ServerApi.prototype, "signIn")
+          .mockImplementation();
 
-    it("should navigate to dashboard on success", async () => {
-      screen.getByTestId("LogInTabButton").click();
+        fireEvent.submit(form);
+        expect(mockSignIn).not.toBeCalled();
+      });
 
-      const form = screen.getByTestId("form");
-      const mockSignIn = jest
-        .spyOn(ServerApi.prototype, "signIn")
-        .mockImplementation(() =>
-          Promise.resolve({ response: {} as AxiosResponse })
-        );
+      it("should not call SignIn if errors are present", async () => {
```

Comment:
I would add a test for a backend response indicating a bad password or something